### PR TITLE
Support GHC 9.4

### LINKED ISSUE #291: Support GHC 9.4
```
$ cabal unpack cborg-0.2.7.0
$ cd cborg-0.2.7.0
$ cabal test -w ghc-9.4.0.20220501 --allow-newer
[ 4 of 14] Compiling Codec.CBOR.Decoding

src/Codec/CBOR/Decoding.hs:331:19: error:
    • Couldn't match expected type ‘Int64#’ with actual type ‘Int#’
    • In the first argument of ‘I64#’, namely ‘n’
      In the expression: I64# n
      In an equation for ‘toInt64’: toInt64 n = I64# n
    |
331 | toInt64  n = I64# n
    |                   ^

src/Codec/CBOR/Decoding.hs:332:19: error:
    • Couldn't match expected type ‘Word64#’ with actual type ‘Word#’
    • In the first argument of ‘W64#’, namely ‘n’
      In the expression: W64# n
      In an equation for ‘toWord64’: toWord64 n = W64# n
    |
332 | toWord64 n = W64# n
    |                   ^

src/Codec/CBOR/Decoding.hs:751:53: error:
    • Couldn't match expected type ‘Word64#’ with actual type ‘Word#’
    • In the first argument of ‘W64#’, namely ‘w#’
      In the first argument of ‘k’, namely ‘(W64# w#)’
      In the expression: k (W64# w#)
    |
751 |   Decoder (\k -> return (ConsumeTag (\w# -> k (W64# w#))))
    |                                                     ^^

src/Codec/CBOR/Decoding.hs:772:62: error:
    • Couldn't match expected type ‘Word64#’ with actual type ‘Word#’
    • In the first argument of ‘W64#’, namely ‘w#’
      In the first argument of ‘k’, namely ‘(W64# w#)’
      In the expression: k (W64# w#)
    |
772 |   Decoder (\k -> return (ConsumeTagCanonical (\w# -> k (W64# w#))))
    |                                                              ^^

src/Codec/CBOR/Decoding.hs:971:74: error:
    • Couldn't match expected type ‘Int64#’ with actual type ‘Int#’
    • In the first argument of ‘I64#’, namely ‘off#’
      In the first argument of ‘k’, namely ‘(I64# off#)’
      In the expression: k (I64# off#)
    |
971 | peekByteOffset = Decoder (\k -> return (PeekByteOffset (\off# -> k (I64# off#))))
    |
```

### PR DESCRIPTION
**Pull request by parsonsmatt**
Fixes #291 

This is a *very* stupid PR, where I use `CPP` and then do `_f` in the 9.4 case to see what to put in. But the tests all pass, so 🎉 🤷🏻

### CODE REVIEW

File: cborg/src/Codec/CBOR/Decoding.hs [hunk 1]
```diff
@@ -769,7 +780,13 @@ decodeTag64Canonical :: Decoder s Word64
...
 decodeTag64Canonical =
 #if defined(ARCH_64bit)
-  Decoder (\k -> return (ConsumeTagCanonical (\w# -> k (W64# w#))))
+  Decoder (\k -> return (ConsumeTagCanonical (\w# -> k (W64#
+#if MIN_VERSION_base(4,17,0)
+    (wordToWord64# w#)
+#else
+    w#
+#endif
+  ))))
```
Comment (axman6) [inline cborg/src/Codec/CBOR/Decoding.hs:789]: Can this be a CPP macro instead? Using #if’s everywhere feels messy to me.

Comment (parsonsmatt) [reply cborg/src/Codec/CBOR/Decoding.hs:789]: This is the limit of my CPP knowledge - I'd love it if there was an easier way!

**Comment by Bodigrim** [comment]
@FinleyMcIlwaine @bgamari any chance to merge and release this please?

**Comment by Bodigrim** [comment]
Thanks @FinleyMcIlwaine, could you please possibly make a Hackage release any time soon?

**Comment by FinleyMcIlwaine** [comment]
@Bodigrim Done, finally 🙂 Apologies for the delay!

**Comment by Bodigrim** [comment]
Huge thanks!

**Pull request merged**